app/controllers/events_controller.rb

```diff
@@ -1,7 +1,8 @@
 class EventsController < ApplicationController
   def index
     if current_user
-      @events = current_user.followed_activity_events.paginate(page: params[:page], per_page: 15)
+      @q = Event.ransack(params[:q])
+      @events = @q.result.paginate(page: params[:page], per_page: 15)
```

Comment:
Why don't you do `@events = Event.ransack(params[:q]).result.paginate(page: params[:page], per_page: 15)`